NET/ReadyToUseUI.Droid/Fragments/FilterListFragment.cs

```diff
@@ -0,0 +1,159 @@
+using Android.Content;
+using Android.Graphics;
+using Android.Views;
+using Google.Android.Material.BottomSheet;
+using IO.Scanbot.Imagefilters;
+using IO.Scanbot.Sdk.Process;
+using ReadyToUseUI.Droid.Listeners;
+using ReadyToUseUI.Droid.Model;
+
+namespace ReadyToUseUI.Droid.Fragments;
+
+public interface IFilterItemAdapter
+{
+    public Context AdapterContext { get; }
+
+    public List<FilterItem> Filters { get; }
+}
+
+public class FilterListFragment : BottomSheetDialogFragment, IFilterItemAdapter
+{
+    public Context AdapterContext => this.Context;
+    public List<FilterItem> Filters => this.filters;
+
+    private readonly List<FilterItem> filters = new List<FilterItem>
+    {
+        new FilterItem("New Filters", true),
+        new FilterItem(typeof(ScanbotBinarizationFilter)),
```

Comment:
@scan-mayank, rather than storing the type and then creating a filter later via GetParametricFilter, you can instead store a delegate inside of FilterItem which creates the filter

Then you just need a title and the delegate in FilterItemSelected

For example, the resulting code might look like:

`new FilterItem(nameof(CustomBinarizationFilter), () => new ScanbotBinarizationFilter())` 

Then you just have to invoke the delegate

Reply:
@matthewrindel thanks for the feedback, added it.